### DISCUSSION TITLE
DBT Versioned Models are invalid assets

### DISCUSSION THREAD

**Discussion by YupeiChen94WiseRock** [dagster-io/dagster#19514] [Q&A]
Hi, how do I get dagster to work with dbt models that are versioned? I am getting an error that a dbt model that is versioned is not a valid asset key. The models build with no issues in dbt. Am only getting this error in Dagster.

dagster_dbt==0.21.9
dbt-postgres==1.7.2

```1706717397801,"dagster._check.CheckError: Invariant failed. Description: Invalid asset dependencies: {AssetKey(['ring', 'dbt', 'mart__aries__well'])} specified in `internal_asset_deps` argument for multi-asset 'assets' on key 'wells'. Each specified asset key must be associated with an input to the asset or produced by this asset. Valid keys: ...```

**Comment by rexledesma** [comment]
Closing as this is a duplicate of an existing issue: https://github.com/dagster-io/dagster/issues/16046. There is a workaround posted in the issue.

**Reply by YupeiChen94WiseRock** [reply]
Just wanted to point out the posted workaround does not completely work.

It resolves my wells model referencing a versioned mart__aries__well model, but it then chokes on mart__aries__well referencing a versioned stg__aries__well model.

stg__aries__well -> mart__aries__well -> wells

After some testing, overwriting below method this way worked. However this has dagster ignore the version on the model and if we had a model with 2 active and deployed versions it would choke again due to multiple resources with the same asset key.

We had to rollback using dbt model versions entirely.

```
def output_name_fn(dbt_resource_props: Mapping[str, Any]) -> str: 
    if dbt_resource_props.get("version") is not None:
        return dbt_resource_props["unique_id"].split(".")[-2].replace("-", "_")
    else:
        return dbt_resource_props["unique_id"].split(".")[-1].replace("-", "_") 
```
